### ISSUE TITLE
Getting unexpected "'{a}' was used before it was declared, which is illegal for '{b}' variables."

### ISSUE THREAD

**Issue by fasttime** [jshint/jshint#2901]
```js
function A()
{
    console.log(text);
}

function B()
{
    console.log(text);
}

const text = 'Hello';

A();
B();
```

This is actually legal ES6 but produces the error E056: "'text' was used before it was declared, which is illegal for 'const' variables."

I believe a warning with a more appropriate message for this case should be fine. Interestingly, the code below does not trigger the error:

```js
function A()
{
    console.log(text);
}

const text = 'Hello';

A();
```

**Comment by nicolo-ribaudo** [comment]
This is a duplicate of #2838. The fix will be released in the next JSHint version (> 2.9.1)

**Comment by jugglinmike** [comment]
Closing as a duplicate of gh-2838.
